File: 2025_07/leetcode2025-07-03.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    char kthCharacter(int k) {
        string w = "a";
        while (w.size()<k){
            string cur;
            for (auto& c : w){
                cur += ('a'+((c-'a')+1)%26);
            }
            w += cur;
        }
        return w[k-1];
    }
};

int main(){
    Solution s;
    int k = 5;
    cout << s.kthCharacter(k) << endl;
}

File: leetcode2025/2025_07/leetcode2025-07-03.py
from string import *

class Solution:
    def kthCharacter(self, k: int) -> str:
        w = "a"
        lowers = ascii_lowercase
        while len(w)<k:
            cur = ""
            for i in w:
                indx = ord(i)-ord('a')
                cur += lowers[(indx+1)%26]
            w += cur
        return w[k-1]

if __name__ == "__main__":
    s = Solution()
    k = 5
    print(s.kthCharacter(k))
